react-app/src/hooks/useSearchParamsObject.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect, useMemo } from "react";
+import qs from "qs";
+import { useLocation, useSearchParams } from "react-router-dom";
+
+/**
+ * A enhanced useSearchParams that allows objects
+ * rather than only as a key string-value only store
+ *
+ * @param defaultParams - params to set when search string is empty
+ */
+const useSearchParamsObject = <T>(
+  defaultParams?: T
+): [searchParams: T, setSearchParams: (v: T | ((prev: T) => T)) => void] => {
+  const location = useLocation();
+  const [_, setSearchParamsString] = useSearchParams();
+
+  const searchParams: T = useMemo(() => {
+    const searchString = location.search.replace("?", "");
```

Comment:
Encountered problem if we used the searchParamsString from `useSearchParams` instead of `location.search`?

Reply:
useSearchParams return key value pairs, and the value is typed string, but we need nested objects in the url search params, thus this custom hook. 

console.log `[...searchParamsString.entries()]` gives this array:
<img width="574" alt="image" src="https://user-images.githubusercontent.com/106721852/175494830-9fe8b9b8-0072-4417-a95d-7f5f5a14b77d.png">